lib/maxmind/geoip2/reader.rb

```diff
@@ -0,0 +1,160 @@
+# frozen_string_literal: true
+
+require 'maxmind/db'
+require 'maxmind/geoip2/errors'
+require 'maxmind/geoip2/model/city'
+require 'maxmind/geoip2/model/country'
+require 'maxmind/geoip2/model/enterprise'
+
+module MaxMind::GeoIP2
+  # Reader is a reader for the GeoIP2/GeoLite2 database format. IP addresses
+  # can be looked up using the database specific methods.
+  #
+  # == Example
+  #
+  #   require 'maxmind/geoip2'
+  #
+  #   reader = MaxMind::GeoIP2::Reader.new('GeoIP2-Country.mmdb')
+  #
+  #   record = reader.country('1.2.3.4')
+  #   puts record.country.iso_code
+  #
+  #   reader.close
+  class Reader
+    # Create a Reader for looking up IP addresses in a GeoIP2/GeoLite2 database
+    # file.
+    #
+    # If you're performing multiple lookups, it's most efficient to create one
+    # Reader and reuse it.
+    #
+    # Once created, the Reader is safe to use for lookups from multiple
+    # threads. It is safe to use after forking only if you use
+    # MaxMind::DB::MODE_MEMORY or if your version of Ruby supports IO#pread.
+    #
+    # Creating the Reader may raise an exception if initialization fails.
+    #
+    # +database+ is a path to a GeoIP2/GeoLite2 database file.
```

Comment:
I don't think you need to change it, but it might have been nice to use YARD for documentation. I think various IDEs and static analyzers can use it for type hints and displaying the documentation inline.

Reply:
Yeah, perhaps that would be a good improvement, but we could do it separately!